src/common/molecules/TErrorModal.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { ReactNode, useState } from "react";
+import TModal from "./TModal";
+import { TButton } from "../atoms/TButton";
+import { TP } from "../atoms/THeadings";
+import {
+  HttpError,
+  EosioUtil,
+  CommunicationError,
+  AntelopePushTransactionError,
+} from "@tonomy/tonomy-id-sdk";
+import { Modal } from "@mui/material";
+
+export type TErrorModalProps = React.ComponentProps<typeof Modal> & {
+  onPress: () => void;
+  title?: string;
+  icon: string;
+  error?: Error;
+  expected?: boolean;
+  open?: boolean;
+  code?: number;
+  cause?: string;
+};
+
+export default function TErrorModal(props: TErrorModalProps) {
+  const [expanded, setExpanded] = useState(false);
+
+  function switchExpanded() {
+    setExpanded((expanded) => !expanded);
+  }
+
+  if (props?.expected === false) {
+    console.error(props.error?.message, JSON.stringify(props.error, null, 2));
+    // TODO: log to Tonomy Foundation team
+  }
+
+  function isExpandableErrorType() {
+    return (
+      props?.error instanceof HttpError ||
+      props?.error instanceof CommunicationError ||
+      props?.error instanceof EosioUtil.AntelopePushTransactionError
+    );
+  }
+
+  function isExpandable() {
+    return isExpandableErrorType() || props?.code || props?.cause;
+  }
+
+  function ErrorDetails() {
+    if (props.error instanceof HttpError) {
+      const error = props.error as HttpError;
+
+      return (
+        <div>
+          <TP size={2}>Http error:</TP>
+          {error.code && (
+            <div style={styles.greyText}>HTTP Code: {error.code}</div>
+          )}
+          <div>Path: {error.path}</div>
+          <div>Response: {JSON.stringify(error.response, null, 2)}</div>
+          <div>SourceUrl: {error.sourceURL}</div>
+        </div>
+      );
+    } else if (props.error instanceof AntelopePushTransactionError) {
+      const error = props.error as AntelopePushTransactionError;
+      const trxError = error.error;
+
+      return (
+        <div>
+          <TP size={2}>Trx error:</TP>
+          {error.code && (
+            <div style={styles.greyText}>HTTP Code: {error.code}</div>
+          )}
+          <div style={styles.greyText}>Antelope Code: {trxError.code}</div>
+          <div style={styles.greyText}>Name: {trxError.name}</div>
+          <div style={styles.greyText}>What: {trxError.what}</div>
+          <div style={styles.greyText}>
```

Comment:
what if we add a parent div and add styles.greyText single time instead of all divs

Reply:
fixed